modules/nextflow/src/main/groovy/nextflow/executor/LsfExecutor.groovy

```diff
@@ -112,6 +112,13 @@ class LsfExecutor extends AbstractGridExecutor {
         return result
     }
 
+    @Override
+    String sanitizeJobName( String name ) {
+        // some implementations do not allow square brackets in the job name -- see #271
+        name = name.replace('[','').replace(']','')
+        // LSF does not allow more than 4094 characters for the job name string
+        name.size()>4094 ? name.substring(0,4094) : name
+    }
```

Comment:
Currently, the superclass trim the name at 256 char. Is it sure that  4094 is the max for \*any* Slurm cluster?

Reply:
The documentation is fairly sparse on job name limits, but I *can* tell you that it depends on the version, not cluster-specific settings, and it has been 4094 for over 10 years. Here is what I posted in the slack:

> I took a look into the incomplete and broken documentation. Maximum job name length seems to be version specific. As far back [as 9.1.1](https://publibfp.dhe.ibm.com/epubs/pdf/c2753051.pdf), it was already 4094, but there is [another document](https://www.ibm.com/support/pages/maximum-number-characters-job-name-can-have) that says the maximum length was 255 until 6.2 and 511 therafter. I have confirmed through testing that the limit is exactly 4094 on LSF 10.1.
>
> So maybe there should be a version check and the limit should be set based on that, or maybe I should change the limit to 255 or 511? I don't know how I would efficiently check the lsf version when extending AbstractGridExecutor (or at all, for that matter; `lsid` gives the version, but it doesn't seem to be consistent or easily parsable).
>
> This might all be beyond the scope of this pull request since the limit has been 4094 for over 10 years.

It's been at least 511 since 2005 if that's preferable.

Reply:
Yes, let's keep at 511 to not break all version

Reply:
I have committed a change to 511